tools/scripts/helpers/issue.py

```diff
@@ -0,0 +1,162 @@
+from dataclasses import dataclass
+from typing import List
+
+from helpers.github import GitHub
+from helpers.utils import extract_url, to_datetime, extract_issue_number
+
+WEBCOMPAT_URL = "https://webcompat.com/issues/"
+
+BROWSER_MAP = {
+    "browser-fenix": "mobile",
+    "browser-firefox-mobile": "mobile",
+    "browser-focus-geckoview": "mobile",
+    "browser-android-components": "mobile",
+    "browser-firefox-ios": "mobile",
+    "browser-firefox": "desktop"
+}
+
+RESOLVED_MAP = {
+    "fixed": "site_fixed",
+    "worksforme": "site_changed",
+    "intervention": "site_changed"
+}
+
+SEVERITY_MAP = {
+    "severity-minor": "minor_visual",
+    "severity-important":
+        "Suggested impact is `feature_broken` or `significant_visual` as issue has severity-important label.",
+    "severity-critical": "Suggested impact is `site_broken` or `feature_broken` as issue has severity-critical label.",
+    "type-unsupported": "unsupported_message"
+}
+
+TRACKED_MILESTONES = [
+    "needsdiagnosis",
+    "duplicate",
+    "moved",
+    "contactready",
+    "needscontact",
+    "sitewait"
+]
+
+
+@dataclass
+class WebcompatIssue:
+    body: str
+    breakage_url: str
+    number: int
+    title: str
+    labels: List[str]
+    milestone: str
+    timeline: List[dict]
+
+    @classmethod
+    def from_dict(cls, issue: dict):
+        original_labels = issue.get("labels", [])
+        labels = [label["name"] for label in original_labels]
+        issue_body = issue["body"]
+        breakage_url = extract_url(issue_body)
+
+        milestone = ""
+        if issue.get("milestone"):
+            milestone = issue["milestone"]["title"]
+
+        return cls(body=issue_body,
+                   breakage_url=breakage_url,
+                   number=issue["number"],
+                   title=issue["title"],
+                   labels=labels,
+                   milestone=milestone,
+                   timeline=issue["timeline"])
+
+    def build_canonical_url(self) -> str:
+        return WEBCOMPAT_URL + str(self.number)
+
+    def get_platform(self) -> list:
+        browser = []
+        for label in self.labels:
+            if label in BROWSER_MAP:
+                browser.append(BROWSER_MAP[label])
+
+        if len(browser) >= 2:
+            browser = ['all']
```

Comment:
Hmm, so this is going to assume that anything that happens on desktop is all operating systems. That isn't always true, although it's probably the common case.